gcs/bucket.py

```diff
@@ -285,6 +309,26 @@ def _init_defaults(cls, metadata: storage_pb2.Bucket, context):
         ).hexdigest()
         metadata.etag = cls._metadata_etag(metadata)
 
+    @classmethod
+    def __validate_soft_delete_policy(cls, metadata, context):
+        if not metadata.HasField("soft_delete_policy"):
+            return
+        policy = metadata.soft_delete_policy
+        if policy.retention_duration.nanos != 0:
+            testbench.error.invalid(
+                "SoftDeletePolicy.retention_duration should not have nanoseconds",
+                context,
+            )
+        if policy.retention_duration.ToSeconds() < 7 * 86400:
+            testbench.error.invalid(
+                "SoftDeletePolicy.retention_duration should be at least 7 days", context
+            )
+        if policy.retention_duration.ToSeconds() >= 365 * 86400:
+            testbench.error.invalid(
+                "SoftDeletePolicy.retention_duration should be at less than a year",
```

Comment:
Is soft delete retention_duration maximum 90 days? This is what I see in the user guide:
`The value must be greater than or equal to 604,800 seconds (7 days) and less than 7,776,000 seconds (90 days)`

nit: extra at within error message